hs/src/Reach/AST/SL.hs

```diff
@@ -160,6 +160,188 @@ data SLVal
   | SLV_Deprecated Deprecation SLVal
   deriving (Eq, Generic)
 
+class NewerVar a where
+  containsVarNewerThan :: Int -> a -> Bool
+
+instance NewerVar SLVal where
+  containsVarNewerThan ret (SLV_DLVar (DLVar _ _ _ i)) = ret < i
+  containsVarNewerThan _ _ = False
```

Comment:
Sadly, this needs to recur inside of all the various SLVals looking for embedded DLVars. For example, if the `SLVal` were a tuple and the second element were a DLVar. So... you need to go searching through 'em all and implement many instances of `NewerVar`.... probably for all the same structures as `Equiv`

Reply:
Any time when `Equiv` would give up, this should say that there IS a variable and thus it fails to unify... i.e. it is a conservative check

Reply:
I added the constructors which had nested SLVals. It wasn't too many of them, so maybe I'm misunderstanding the tree we should be inspecting